doc/sphinx/source/developer_guide/logging.rst

```diff
@@ -0,0 +1,264 @@
+
+.. code-block:: xml
+
+  <firmware name="rotorcraft">
+    <module name="tlsf"/>
+    <module name="pprzlog"/>
+    <module name="logger" type="sd_chibios"/>
+    <module name="flight_recorder"/>
+  </firmware>
+
+Data is stored on an SD card using the ``pprzlog`` format, discussed in more detail in the next section.
+The logging starts automatically soon after the drone is connected to power. The logger status is reported
+on the ``LOGGER_STATUS`` message, which can be viewed using the GCS by opening Tools -> Messages.
+
+The ``flight_recorder`` will record all messages that are specified within a ``FlightRecorder`` process
+that should be included in your airframe's telemetry file.
+
+.. code-block:: xml
+
+  <telemetry>
+    <process name="FlightRecorder">
+      <mode name="default">
+        <message name="ATTITUDE"    period="0.05"/>
+        <message name="IMU_ACCEL"   period="0.02"/>
+        <message name="IMU_GYRO"    period="0.02"/>
+        <message name="IMU_MAG"     period="0.02"/>
+        <!-- etc. -->
+      </mode>
+    </process>
+  </telemetry>
+
+These telemetry files are located within the ``conf/telemetry/`` folder, and are associated to a specific
+airframe in the ``conf.xml``. For example, ``conf/userconf/tudelft/conf.xml`` specifies for a default Bebop:
+
+.. code-block:: xml
+
+  <conf>
+    <aircraft
+     name="Bebop_default"
+     ac_id="20"
+     airframe="airframes/examples/bebop.xml"
+     radio="radios/dummy.xml"
+     telemetry="telemetry/default_rotorcraft.xml"
+     flight_plan="flight_plans/rotorcraft_basic.xml"
+     settings="settings/rotorcraft_basic.xml settings/control/rotorcraft_speed.xml"
+     settings_modules="modules/ahrs_float_mlkf.xml modules/air_data.xml modules/bebop_ae_awb.xml modules/bebop_cam.xml modules/geo_mag.xml modules/gps.xml modules/guidance_rotorcraft.xml modules/imu_common.xml modules/ins_extended.xml modules/nav_basic_rotorcraft.xml modules/stabilization_int_quat.xml modules/video_rtp_stream.xml"
+     gui_color="#ffffbc3bce5b"
+    />
+  </conf>
+
+Decoding FlightRecorder logs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To download and convert the data, you need to first connect the SD card to your laptop, navigate to the
+FLIGHT_RECORDER folder and transfer to your computer the relevant logs, named ``fr_XXXX.LOG``. To convert
+this binary file into Pprzlog format files the program ``sw/logalizer/sd2log`` has to be used.
+Make sure the environment variables are set before running Paparazzi executables from the commandline.
+They can be set in your Terminal by using
+
+.. code-block:: bash
+
+  export PAPARAZZI_HOME=~/paparazzi
+  export PAPARAZZI_SRC=~/paparazzi
+
+.. tip::
+
+  You can add the ``export`` lines above to your ``.bashrc`` file to automatically set the environment variables
+  every time a new Terminal is opened.
+
+Once the environment variables are set you can run ``sd2log`` in your terminal:
+
+.. code-block:: bash
+
+  ~/paparazzi/sw/logalizer/sd2log fr_XXXX.LOG
```

Comment:
`~/paparazzi/sw/logalizer/sd2log fr_XXXX.LOG output_dir_path` to extract the log in an other folder than default log path